lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4464,11 +4464,15 @@ namespace Js
         }
 
         // fastpath for Symbols only if receiver == object
-        JavascriptSymbol * symbol = JavascriptOperators::TryFromVar<JavascriptSymbol>(index);
+        JavascriptSymbol * symbol = nullptr;
 
-        if (symbol != nullptr)
+        if (propertyString != nullptr)
```

Comment:
Should this check be `propertyString == nullptr`? I can see how we wouldn't want to waste time with `TryFromVar<JavascriptSymbol>(index)` if we already know that `index` is a property string, but property strings and symbols are separate classes of things so `index` can't be both.

Reply:
Here's a simple test script I used to make sure we were caching symbol get/set operations correctly. On my machine, it finishes in around 350 ms with Symbol caching and over 25 seconds (at least on debug build) if Symbol caching is disabled.

```javascript
const str = "hello";
const sym = Symbol();

function getStringProp(a) {
    return a[str];
}

function getSymbolProp(b) {
    return b[sym];
}

function setStringProp(a, val) {
    a[str] = val;
}

function setSymbolProp(b, val) {
    b[sym] = val;
}

class A1 {
    constructor(val) {
        this[str] = val;
    }
}

class A2 {
    constructor(val) {
        this.foo = "hi";
        this[str] = val;
    }
}

class B1 {
    constructor(val) {
        this[sym] = val;
    }
}

class B2 {
    constructor(val) {
        this.foo = "hi";
        this[sym] = val;
    }
}

const a1 = new A1(2);
const a2 = new A2(3);
const b1 = new B1(1);
const b2 = new B2(4);

const now = Date.now();
let counter = 0;

for (let i = 0; i < 1e7; ++i) {
    counter += getStringProp(a1);
    counter += getStringProp(a2);
    counter += getSymbolProp(b1);
    counter += getSymbolProp(b2);
    setStringProp(a1, 2);
    setStringProp(a2, 3);
    setSymbolProp(b1, 1);
    setSymbolProp(b2, 4);
}

print(Date.now() - now);

print(counter);
```

Reply:
> `Should this check be propertyString == nullptr?`

It definitely should. It's a typo.